message/pool/message.go

```diff
@@ -81,7 +81,7 @@ func (r *Message) SetMessageID(mid int32) {
 
 // UpsertMessageID set value only when origin value is invalid. Only 0 to 2^16-1 values are valid.
 func (r *Message) UpsertMessageID(mid int32) {
-	if r.msg.MessageID >= 0 && r.msg.MessageID < math.MaxUint16 {
+	if r.msg.MessageID >= 0 && r.msg.MessageID <= math.MaxUint16 {
```

Comment:
Please, could you create `message.ValidateMessageID(mid int32) bool`, it will be used at both places, as asked for by @mniestroj?